src/ImportDefinitionsBundle/Export/Exporter.php

```diff
@@ -0,0 +1,266 @@
+    private $exceptions = [];
+
+    /**
+     * Importer constructor.
+     * @param ServiceRegistryInterface $fetcherRegistry
+     * @param ServiceRegistryInterface $runnerRegistry
+     * @param ServiceRegistryInterface $reverseInterpreterRegistry
+     * @param ServiceRegistryInterface $getterRegistry
+     * @param EventDispatcherInterface $eventDispatcher
+     * @param LoggerInterface $logger
+     */
+    public function __construct(
+        ServiceRegistryInterface $fetcherRegistry,
+        ServiceRegistryInterface $runnerRegistry,
+        ServiceRegistryInterface $reverseInterpreterRegistry,
+        ServiceRegistryInterface $getterRegistry,
+        EventDispatcherInterface $eventDispatcher,
+        LoggerInterface $logger
+    )
+    {
+        $this->fetcherRegistry = $fetcherRegistry;
+        $this->runnerRegistry = $runnerRegistry;
+        $this->reverseInterpreterRegistry = $reverseInterpreterRegistry;
+        $this->getterRegistry = $getterRegistry;
+        $this->eventDispatcher = $eventDispatcher;
+        $this->logger = $logger;
+    }
+
+    /**
+     * {@inheritdoc}
+     * @throws \Exception
+     */
+    public function doExport(DefinitionInterface $definition, $params)
+    {
+        $fetcher = $this->getFetcher($definition);
+        $total = $fetcher->count($definition, $params);
+
+        if ($total > 0) {
+            $this->eventDispatcher->dispatch('export_definition.total', new ImportDefinitionEvent($definition, $total));
+
+            $this->runExport($definition, $params, $total, $fetcher);
+        }
+
+        $this->eventDispatcher->dispatch('export_definition.finished', new ImportDefinitionEvent($definition));
+    }
+
+    /**
+     * @param DefinitionInterface $definition
+     * @return FetcherInterface
+     */
+    private function getFetcher(DefinitionInterface $definition)
+    {
+        if (!$this->fetcherRegistry->has($definition->getFetcher())) {
+            throw new \InvalidArgumentException(sprintf('Definition %s has no valid fetcher configured', $definition->getName()));
+        }
+
+        /** @var FetcherInterface $fetcher */
+        $fetcher = $this->fetcherRegistry->get($definition->getFetcher());
+
+        return $fetcher;
+    }
+
+    /**
+     * @param DefinitionInterface $definition
+     * @param                     $params
+     * @param int                 $total
+     * @param FetcherInterface    $fetcher
+     * @throws \Exception
+     */
+    private function runExport(DefinitionInterface $definition, $params, int $total, FetcherInterface $fetcher)
+    {
+        $count = 0;
+        $countToClean = 1000;
+        $perLoop = 50;
+
+        $entries = [];
+
+        for ($i = 0; $i < (ceil($total / $perLoop)); $i++) {
+
+            $objects = $fetcher->fetch($definition, $params, $perLoop, $i * $perLoop);
+
+            foreach ($objects as $object) {
+                try {
+                    $entries[] = $this->exportRow($definition, $object, $params);
```

Comment:
This keeps collecting the data in memory. A better approach would be to stream the data into a file, using for example https://github.com/box/spout

Reply:
right... good idea, but these somehow needs to be more abstract than just streaming, since the output channel could be anything.

Reply:
Maybe add another interface "StreamingExportProviderInterface", or simply add a method "isStreaming" to the ExportProviderInterface. Then in exporter export one entry at a time if it is streaming. Otherwise collect entries like now and export everything at the end.